### DISCUSSION TITLE
Can I use remix js for only frontend and for only backend?

### DISCUSSION THREAD

**Discussion by Abduraimbek** [remix-run/remix#4641] [Q&A] [answered]
I am new to remix js and I want to learn remix js. I am confused that can I use remix js for only frontend and for only backend?

**Comment by sergiodxa** [comment, marked as answer]
You can use Remix as an API if you want yes, that wouldn't be too much different than Express since Remix gives you the router as Express and a few more things like a way to parse the body.

You can also use Remix as your frontend consuming an API, that's how I use it on production, but remember that even if Remix is only your frontend it still needs a server, so it's not a frontend in the traditional SPA way, with code running 100% on the browser, your loaders and actions still run server-side, you still have SSR, it just change that instead of querying your DB directly you fetch an API.

If you want something like Remix but 100% client-side, the latest version of React Router implements the same features (loaders, actions, form, useFetcher, etc.) and a few more things coming to Remix soon, but it lets you build a 100% client-side SPA using something like Vite or Create React App or any other bundler.

**Reply by Abduraimbek** [reply]
Thanks for reply!
